logprep/util/grok/grok.py

```diff
@@ -79,26 +79,35 @@ def __attrs_post_init__(self):
     def match(self, text):
         """If text is matched with pattern, return variable names
         specified(%{pattern:variable name}) in pattern and their
-        corresponding values.If not matched, return None.
+        corresponding values. If not matched, return None.
         custom patterns can be passed in by
         custom_patterns(pattern name, pattern regular expression pair)
         or custom_patterns_dir.
         """
 
         if self.fullmatch:
-            match_obj = self.regex_obj.fullmatch(text)
+            match_obj = list(map(lambda x: x.fullmatch(text), self.regex_obj))
```

Comment:
I think a list comprehension should be faster here because of the additional converting to a list.

And it will be more readable in my opinion

Reply:
I revised it